commonmark/src/main/java/org/commonmark/parser/Parser.java

```diff
@@ -75,12 +77,20 @@ public Node parse(String input) {
      * @throws IOException when reading throws an exception
      */
     public Node parseReader(Reader input) throws IOException {
-        InlineParserImpl inlineParser = new InlineParserImpl(specialCharacters, delimiterCharacters, delimiterProcessors);
+        InlineParser inlineParser = getInlineParser();
         DocumentParser documentParser = new DocumentParser(blockParserFactories, inlineParser);
         Node document = documentParser.parse(input);
         return postProcess(document);
     }
 
+    private InlineParser getInlineParser() {
+        if (this.inlineParserFactory == null) {
+            return new InlineParserImpl(specialCharacters, delimiterCharacters, delimiterProcessors);
+        } else {
+            return this.inlineParserFactory.create();
```

Comment:
Does your custom inline parser not need to know about `delimiterProcessors` etc? I think the `create` method should get a `InlineParserContext` (similar to `HtmlNodeRendererContext`), that looks like this:

```java
interface InlineParserContext {
    Map<Character, DelimiterProcessor> getDelimiterProcessors();
}
```

(The other two, `specialCharacters` and `delimiterCharacters` can be derived from the delimiter processors and are just an optimization, so don't need to be on the interface.)